frontend/lib/activation/deeplink_activation.dart

```diff
@@ -0,0 +1,211 @@
+import 'dart:convert';
+
+import 'package:ehrenamtskarte/app.dart';
+import 'package:ehrenamtskarte/build_config/build_config.dart' show buildConfig;
+import 'package:ehrenamtskarte/configuration/configuration.dart';
+import 'package:ehrenamtskarte/graphql/graphql_api.graphql.dart';
+import 'package:ehrenamtskarte/home/home_page.dart';
+import 'package:ehrenamtskarte/identification/activation_workflow/activate_code.dart';
+import 'package:ehrenamtskarte/identification/activation_workflow/activation_code_parser.dart';
+import 'package:ehrenamtskarte/identification/activation_workflow/activation_exception.dart';
+import 'package:ehrenamtskarte/identification/qr_code_scanner/qr_code_processor.dart';
+import 'package:ehrenamtskarte/identification/qr_code_scanner/qr_parsing_error_dialog.dart';
+import 'package:ehrenamtskarte/identification/user_code_model.dart';
+import 'package:ehrenamtskarte/identification/util/card_info_utils.dart';
+import 'package:ehrenamtskarte/l10n/translations.g.dart';
+import 'package:ehrenamtskarte/proto/card.pb.dart';
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+import 'package:graphql_flutter/graphql_flutter.dart';
+import 'package:provider/provider.dart';
+import 'package:ehrenamtskarte/identification/id_card/id_card.dart';
+import 'package:ehrenamtskarte/graphql/graphql_api.dart';
+
+enum DeepLinkActivationStatus {
+  // Link is invalid
+  invalidLink,
+  // The card already exists on the device.
+  alreadyExists,
+  // The card limit is reached on the device.
+  limitReached,
+  // The card can be activated
+  valid;
+
+  factory DeepLinkActivationStatus.from(UserCodeModel userCodeModel, DynamicActivationCode? activationCode) {
+    if (activationCode == null) {
+      return DeepLinkActivationStatus.invalidLink;
+    } else if (isAlreadyInList(userCodeModel.userCodes, activationCode.info)) {
+      return DeepLinkActivationStatus.alreadyExists;
+    } else if (hasReachedCardLimit(userCodeModel.userCodes)) {
+      return DeepLinkActivationStatus.limitReached;
+    } else {
+      return DeepLinkActivationStatus.valid;
+    }
+  }
+}
+
+class DeepLinkActivation extends StatelessWidget {
+  final String base64qrcode;
```

Comment:
(optional/minor) I would prefer qrCodeBase64, as the more important information (that it is a qrCode) is at the beginning